src/node_zlib.cc

```diff
@@ -236,6 +239,40 @@ class ZCtx : public AsyncWrap {
         ctx->err_ = deflate(&ctx->strm_, ctx->flush_);
         break;
       case UNZIP:
+        next_expected_header_byte = ctx->strm_.next_in;
+
+        switch (ctx->gzip_id_bytes_read_) {
```

Comment:
In that vein, shouldn't there be a similar check for GZIP_HEADER_ID2?

Another question: what happens when the state machines sees GZIP_HEADER_ID1 followed by a byte that's not GZIP_HEADER_ID2?  Is the first byte lost?